### PR TITLE
Fixing jinja2 warnings, ansiblelint and yamllint errors

### PR DESCRIPTION
**Pull request by imjoseangel**
I have fix all the jinja2 warnings and adapt the code in a more Ansible way. Code has been indented properly and linted to pass all the ansible and yamllint test except line lenght. New .yamllint file has been added in root to avoid future warnings.

### CODE REVIEW

File: tasks/parameter/sp.yml [hunk 1]
```diff
@@ -1,28 +1,29 @@
...
 
 - name: Load service_princial in ~/.azure/credentials
   set_fact:
-      service_principal: "{{ lookup('ini', 'client_id section=default file=~/.azure/credentials') }}"
-      client_secret: "{{ lookup('ini', 'secret section=default file=~/.azure/credentials') }}"
-  when: service_principal is not defined or service_principal == ""
-  ignore_errors: yes
+    service_principal: "{{ lookup('ini', 'client_id section=default file=~/.azure/credentials') }}"
+    client_secret: "{{ lookup('ini', 'secret section=default file=~/.azure/credentials') }}"
+  when: (service_principal is undefined) or (service_principal is defined and not service_principal | trim | length > 0)
```
Comment (mmyyrroonn) [inline tasks/parameter/sp.yml:20]: Hello, I am just curious about the benefit of this change.

File: .yamllint [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+extends: default
+
+rules:
+  line-length: disable
```
Comment (mmyyrroonn) [inline .yamllint:5]: Is there any way to set the line-length instead of just disabling this rule?

**Review by mmyyrroonn — APPROVED** [review]

**Comment by mmyyrroonn** [comment]
LGTM. Most of the change are whitespace changes. Just a few questions to learn about.

File: tasks/parameter/sp.yml (hunk 1 above)
Comment (imjoseangel) [reply tasks/parameter/sp.yml:20]: No benefit regarding code. Just following best practices (See ansiblelint at https://github.com/ansible/ansible-lint)

File: .yamllint (hunk 2 above)
Comment (imjoseangel) [reply .yamllint:5]: Yep, we can adjust to the desired length but better to disable as there are commands or paths that can break the rule (Not at this role but also good to have)

**Comment by imjoseangel** [comment]
@MyronFanQiu are you able to merge the PR?

Thanks!

**Comment by mmyyrroonn** [comment]
@imjoseangel  Yes. I can merge it. Sorry about merging another PR first. please solve the conflict and I will merge it. 😄

**Comment by imjoseangel** [comment]
@MyronFanQiu Thanks! Fixed!

**Comment by mmyyrroonn** [comment]
@imjoseangel Hello, it seems the latest commit just revert your design. Is that correct?

**Comment by imjoseangel** [comment]
@MyronFanQiu  Nope, I only missed ---, but the rest is OK

**Comment by mmyyrroonn** [comment]
Thanks for your contribution!

**Pull request merged**